### PR TITLE
Add capital gains basis indexation reform

### PR DESCRIPTION
**Pull request by MaxGhenis**
## Summary

Adds the PolicyEngine-US side of capital gains basis indexation, stacked on #8423's new basis and holding-period inputs.

This PR adds:

- IRS capital gains indexation parameters for activation, minimum holding period, acquisition-year threshold, no-loss cap, and chained-CPI transition year
- a tax-unit-level basis indexation adjustment, computed from the collapsed tax-unit representation
- person-level allocation of the tax-unit adjustment so existing person-level `long_term_capital_gains` consumers continue to work
- prospective, retrospective, and retrospective-losses-allowed reform helpers for benchmark comparisons
- YAML tests covering inactive baseline behavior, prospective/retrospective eligibility, no-loss capped gains, mixed-spouse collapsed tax-unit behavior, and the losses-allowed Yale benchmark axis

## Notes

This intentionally computes the adjustment at the tax-unit level first. The data imputation PR stores basis and holding period at person level for compatibility, but the collapsed imputation is conceptually one representative transaction per tax unit.

The default reform helpers use a no-loss cap. A separate `capital_gains_indexation_retrospective_losses_allowed` helper is included for benchmarking against the public Yale runscript, which appears to leave `index_kg_no_loss = 0`.

Stacked on #8423. Related data issue: PolicyEngine/policyengine-us-data#1126.

## Tests

- `uv run ruff check policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/tax_unit_long_term_capital_gains_indexation_adjustment.py policyengine_us/variables/household/income/person/capital_gains/long_term_capital_gains_indexation_adjustment.py policyengine_us/reforms/capital_gains_indexation.py policyengine_us/variables/household/income/person/capital_gains/long_term_capital_gains.py`
- `uv run python -m policyengine_core.scripts.policyengine_command test policyengine_us/tests/policy/contrib/capital_gains_indexation.yaml -c policyengine_us`
- `uv run python -m policyengine_core.scripts.policyengine_command test policyengine_us/tests/policy/baseline/gov/irs/tax/federal_income/capital_gains policyengine_us/tests/policy/baseline/gov/irs/integration/capital_gains_losses.yaml policyengine_us/tests/variables/household/income/person/capital_gains -c policyengine_us`
- `uv run pytest policyengine_us/tests/test_parameter_files.py policyengine_us/tests/test_system_import.py -q`

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Closing as superseded (2026-07-04 worktree/PR sweep).

Evidence: Not superseded by equivalent landed code, but its entire dependency chain was formally killed. Tracking issue PolicyEngine/policyengine-us-data#1126 ("Add collapsed SOCA imputation for capital gains basis and holding period") and its implementation PR policyengine-us-data#1128 were both closed 2026-07-02 with the identical comment: "Closing as part of archiving this repository — Populace (PolicyEngine/populace) is its replacement and the certified US dataset since June 2026; see PolicyEngine/populace#204 ... Published artifacts remain available but frozen." The policyengine-us-data repo itself is archived (archivedAt: 2026-07-02T11:45:32Z). The sibling synthetic-lots issue #1127 is also closed. The successor tracking issue is populace#38 ("Populate remaining US tax-input and reported-obser

The head branch and a rescue of its worktree state are preserved locally and in git bundles under ~/.claude/backups/worktree-sweep-20260704/. Reopen if this is wrong.
